capture.js

```diff
@@ -0,0 +1,47 @@
+const capture = {
+
+  init() {
+    this.addListeners();
+  },
+
+  addListeners() {
+    // listen for each HTTP response
+    browser.webRequest.onResponseStarted.addListener(this.collectRequests, {urls: ['<all_urls>']}
+    );
+  },
+
+  /*
+  * makes calls to the storage API for each
+  * first and third party request
+  */
+  async collectRequests(response) {
+    const tab = await browser.tabs.get(response.tabId);
+    let tabUrl = new URL(tab.url);
+    tabUrl = tabUrl.hostname;
+    let targetUrl = new URL(response.url);
+    targetUrl = targetUrl.hostname;
+    let originUrl = new URL(response.originUrl);
+    originUrl = originUrl.hostname;
+
+    //capture first party requests
+    const re = /^about:/;
+    // ignore non-visible tabs; i.e. about:* pages
+    if (tab.status === 'complete' && !(re.test(tabUrl))) {
+      const firstPartyData = {
+        faviconUrl: tab.favIconUrl,
+      };
+      console.log('storage.firstPartyRequest:', tabUrl, firstPartyData);
+    }
+
+    // capture third party requests
+    if (targetUrl !== originUrl) {
```

Comment:
I need to check if this is true... I haven't tested this yet. originUrl is where it loads from rather than the top level frame URL from my testing. This would treat the first party being loaded by a third party as another third party I suspect.
So:

- google.com
  - other.com/tracking.js
    - google.com/callback.js << considered a third party